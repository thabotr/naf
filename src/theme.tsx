import AsyncStorage from '@react-native-async-storage/async-storage';

export function getSavedThemeSetting(): Promise<string|null>{
  return AsyncStorage.getItem('@naf_theme_setting');
}

export function saveThemeSetting(themeSetting: 'light' | 'dark' | 'system_default'): Promise<void> {
  return AsyncStorage.setItem('@naf_theme_setting', themeSetting);
}