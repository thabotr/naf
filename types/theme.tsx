export type ThemeType = {
  dark: boolean;
  color: {
    primary: string;
    secondary: string;
    userPrimary: string;
    friendPrimary: string;
    userSecondary: string;
    friendSecondary: string;
  }
};

export type ThemeContextType = {
  theme: ThemeType;
  toggleDarkTheme: ()=>void;
};