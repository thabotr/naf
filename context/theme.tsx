import React from 'react';
import {useColorScheme} from 'react-native';
import { getSavedThemeSetting, saveThemeSetting as storageSaveThemeSetting } from '../src/theme';

export type ThemeType = {
  dark: boolean;
  color: {
    primary: string;
    secondary: string;
    userPrimary: string;
    friendPrimary: string;
    userSecondary: string;
    friendSecondary: string;
    textPrimary: string;
    textSecondary: string;
  }
};

export type ThemeContextType = {
  theme: ThemeType;
  themeSetting: 'light' | 'dark' | 'system_default';
  saveThemeSetting: (ts: 'light' | 'dark' | 'system_default')=>void;
};

export const ThemeContext = React.createContext<ThemeContextType|null>(null);

export function ThemeContextProvider({children}:{children: React.ReactNode}){
  const darkTheme: ThemeType = {
    dark: true,
    color: {
      primary: '#1d2126',
      secondary: '#363636',
      userPrimary: '#404040',
      userSecondary: '#5c5c5c',
      friendPrimary: '#2a4365',
      friendSecondary: '#18474e',
      textPrimary: '#f8f8ff',
      textSecondary: '#dcdcdc',
    }
  }

  const lightTheme: ThemeType = {
    dark: false,
    color: {
      primary: '#87cefa',
      secondary: '#dbdbdb',
      userPrimary: '#c0c0c0',
      userSecondary: '#f5f5f5',
      friendPrimary: '#b0c4de',
      friendSecondary: '#b0e0e6',
      textPrimary: '#000000',
      textSecondary: '#a9a9a9',
    }
  }

  const systemThemeIsDark = useColorScheme() === 'dark';

  const [theme, setTheme] = React.useState<ThemeType>(darkTheme);

  const [themeSetting, setThemeSetting] = React.useState<'light' | 'dark' | 'system_default'>('system_default');

  React.useEffect(()=>{
    getSavedThemeSetting().then(s => {
      switch(s){
        case 'light':
          setTheme(lightTheme);
          setThemeSetting('light');
          return;
        case 'dark':
          setTheme(darkTheme);
          setThemeSetting('dark');
          return;
        default:
          if(systemThemeIsDark) setTheme(darkTheme);
          else setTheme(lightTheme);
          setThemeSetting('system_default');
      }
    })
  },[]);

  const saveThemeSetting = ( ts: 'light' | 'dark' | 'system_default')=> {
    switch(ts){
      case 'light':
        setTheme(lightTheme);
        break;
      case 'dark':
        setTheme(darkTheme);
        break;
      default:
        if(systemThemeIsDark) setTheme(darkTheme);
        setTheme(lightTheme);
    }
    setThemeSetting(ts);
    storageSaveThemeSetting(ts).catch(e=>console.error(e));
  }

  return <ThemeContext.Provider
    value={{
      theme:theme,
      saveThemeSetting: saveThemeSetting,
      themeSetting: themeSetting,
      }}>
    {children}
  </ThemeContext.Provider>
}