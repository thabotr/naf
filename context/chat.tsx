import React from 'react';
import { MessageThread } from '../types/message';
import { Chat } from '../types/chat';
import { Message } from './messageEditor';
import { User } from '../types/user';

export type ChatContextType = {
  chats: Chat[],
  saveChats: (cs: Chat[])=>void,
  conversingWith?: User,
  converseWith: (user: User)=>void,
}

export const ChatContext = React.createContext<ChatContextType|null>(null);

export function ChatContextProvider({children}:{children: React.ReactNode}){
  const [chats, setChats] = React.useState<Chat[]>([]);
  const [conversingWith, setConversingWith] = React.useState<User|undefined>();
  const saveChats = (cs: Chat[])=> setChats(cs);

  const converseWith = (h: User) => setConversingWith(h);

  return <ChatContext.Provider
    value={{
      chats: chats,
      saveChats: saveChats,
      conversingWith: conversingWith,
      converseWith: converseWith,
    }}
  >
    {children}
  </ChatContext.Provider>
}