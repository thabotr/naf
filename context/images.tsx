import React from 'react';
import {ImageViewContextType, Image} from '../types/images';

export const ImageViewContext = React.createContext<ImageViewContextType|null>(null);

export const ImageViewProvider = ({children}:{children: React.ReactNode}) => {
    const [images, setImages] = React.useState<Image[]>([]);
    const [onView, setOnView] = React.useState(false);
    const [sender, setSender] = React.useState(true);

    const saveImages = (images: Image[]) => setImages(images);
    const onViewOff = ()=> setOnView(false);
    const onViewOn = ()=> setOnView(true);
    const saveSender = (b: boolean) => setSender(b);

    return <ImageViewContext.Provider value={{sender, saveSender, images, saveImages, onView, onViewOn, onViewOff}}>
        {children}
    </ImageViewContext.Provider>
}