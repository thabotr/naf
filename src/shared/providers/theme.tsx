import React, {createContext, useState, ReactNode, useContext} from 'react';
import {useColorScheme} from 'react-native';
import {validateContext} from '../utils/validateContext';
import {ThemeSetting} from '../../types/settings';

export type ThemeType = {
  dark: boolean;
  color: {
    primary: string;
    secondary: string;
    userPrimary: string;
    friendPrimary: string;
    userSecondary: string;
    friendSecondary: string;
    textPrimary: string;
    textSecondary: string;
  };
};

export type ThemeContextType = {
  theme: ThemeType;
  saveThemeFromSetting: (ts: ThemeSetting) => void;
  themeSetting: ThemeSetting;
};

export const ThemeSettingFromString: {[key: string]: ThemeSetting} = {
  light: 'light',
  dark: 'dark',
  system_default: 'system_default',
};

const darkTheme: ThemeType = {
  dark: true,
  color: {
    primary: '#1d2126',
    secondary: '#363636',
    userPrimary: '#404040',
    userSecondary: '#5c5c5c',
    friendPrimary: '#2a4365',
    friendSecondary: '#18474e',
    textPrimary: '#f8f8ff',
    textSecondary: '#dcdcdc',
  },
};
const lightTheme: ThemeType = {
  dark: false,
  color: {
    primary: '#87cefa',
    secondary: '#dbdbdb',
    userPrimary: '#c0c0c0',
    userSecondary: '#f5f5f5',
    friendPrimary: '#b0c4de',
    friendSecondary: '#b0e0e6',
    textPrimary: '#000000',
    textSecondary: '#a9a9a9',
  },
};

const ThemeContext = createContext<ThemeContextType | undefined>(undefined);

function ThemeProvider({children}: {children: ReactNode}) {
  const systemTheme = useColorScheme();
  const isSystemDark = systemTheme === 'dark';
  const themeForSetting = (ts: ThemeSetting) => {
    switch (ts) {
      case 'light':
        return lightTheme;
      case 'dark':
        return darkTheme;
      default:
        return isSystemDark ? darkTheme : lightTheme;
    }
  };
  const [themeSetting, setThemeSetting] =
    useState<ThemeSetting>('system_default');
  const saveThemeFromSetting = (ts: ThemeSetting) => {
    setThemeSetting(ts);
  };

  const providerValue = {
    theme: themeForSetting(themeSetting),
    saveThemeFromSetting: saveThemeFromSetting,
    themeSetting: themeSetting,
  };

  return (
    <ThemeContext.Provider value={providerValue}>
      {children}
    </ThemeContext.Provider>
  );
}

const useTheme = (): ThemeContextType => {
  const context = useContext(ThemeContext);
  return validateContext(context, 'useTheme', 'ThemeProvider');
};

const useThemedStyles = (styles: (theme: ThemeType) => any) => {
  const {theme} = useTheme();
  return styles(theme);
};

export {useTheme, ThemeProvider, useThemedStyles};
