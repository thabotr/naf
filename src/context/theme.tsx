import React, {
  createContext,
  useState,
  ReactNode,
  useContext,
  useCallback,
} from 'react';
import {useColorScheme} from 'react-native';
import {validateContext} from '../providers/validateContext';
import {ThemeSetting} from '../types/settings';

export type ThemeType = {
  dark: boolean;
  color: {
    primary: string;
    secondary: string;
    userPrimary: string;
    friendPrimary: string;
    userSecondary: string;
    friendSecondary: string;
    textPrimary: string;
    textSecondary: string;
  };
};

export type ThemeContextType = {
  theme: ThemeType;
  saveThemeFromSetting: (ts: ThemeSetting) => void;
  themeSetting: ThemeSetting;
};

export const ThemeSettingFromString: {[key: string]: ThemeSetting} = {
  light: 'light',
  dark: 'dark',
  system_default: 'system_default',
};

const darkTheme: ThemeType = {
  dark: true,
  color: {
    primary: '#1d2126',
    secondary: '#363636',
    userPrimary: '#404040',
    userSecondary: '#5c5c5c',
    friendPrimary: '#2a4365',
    friendSecondary: '#18474e',
    textPrimary: '#f8f8ff',
    textSecondary: '#dcdcdc',
  },
};
const lightTheme: ThemeType = {
  dark: false,
  color: {
    primary: '#87cefa',
    secondary: '#dbdbdb',
    userPrimary: '#c0c0c0',
    userSecondary: '#f5f5f5',
    friendPrimary: '#b0c4de',
    friendSecondary: '#b0e0e6',
    textPrimary: '#000000',
    textSecondary: '#a9a9a9',
  },
};

const ThemeContext = createContext<ThemeContextType | undefined>(undefined);

function ThemeProvider({children}: {children: ReactNode}) {
  const [themeSetting, setThemeSetting] =
    useState<ThemeSetting>('system_default');
  const isSystemDark = useColorScheme() === 'dark';
  const [theme, setTheme] = useState<ThemeType>(() => {
    if (
      (themeSetting === 'system_default' && !isSystemDark) ||
      themeSetting === 'light'
    ) {
      return lightTheme;
    }
    return darkTheme;
  });

  const themeFromSetting = useCallback(
    (ts: ThemeSetting) => {
      switch (ts) {
        case 'light':
          return lightTheme;
        case 'dark':
          return darkTheme;
        default:
          return isSystemDark ? darkTheme : lightTheme;
      }
    },
    [isSystemDark],
  );

  const saveThemeFromSetting = (ts: ThemeSetting) => {
    setTheme(themeFromSetting(ts));
    setThemeSetting(ts);
  };

  const providerValue = {
    theme: theme,
    saveThemeFromSetting: saveThemeFromSetting,
    themeSetting: themeSetting,
  };

  return (
    <ThemeContext.Provider value={providerValue}>
      {children}
    </ThemeContext.Provider>
  );
}

const useTheme = (): ThemeContextType => {
  const context = useContext(ThemeContext);
  return validateContext(context, 'useTheme', 'ThemeProvider');
};

const useThemedStyles = (styles: (theme: ThemeType) => any) => {
  const {theme} = useTheme();
  return styles(theme);
};

export {useTheme, ThemeProvider, useThemedStyles};
