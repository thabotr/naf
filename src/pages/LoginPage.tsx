import React, {useEffect, useState} from 'react';
import {StyleSheet, View} from 'react-native';
import {Button, Paragraph, Surface, TextInput} from 'react-native-paper';

import {OnlyShow} from '../components/Helpers/OnlyShow';

import {ThemeType, useThemedStyles} from '../shared/providers/theme';

type Props = {
  userCredentials: {
    token: string;
    handle: string;
  };
  loginError?: 'NET_ERROR' | 'AUTH_ERROR';
  onPressLoginBtn: (userCredentials: {token: string; handle: string}) => void;
};

export function Login(props: Props) {
  const styles = useThemedStyles(styleSheet);
  const [loginError, setLoginError] = useState(() => props.loginError);
  const [loading, setLoading] = useState(false);
  const [handle, setHandle] = useState(props.userCredentials.handle);
  const [token, setToken] = useState(props.userCredentials.token);
  const [loginBtnDisabled, disableLoginBtn] = useState(true);

  useEffect(() => {
    if (!handle || !token || loading) {
      disableLoginBtn(true);
    } else {
      disableLoginBtn(false);
    }
  }, [handle, token, loading]);

  useEffect(() => {
    if (props.loginError) {
      setLoginError(props.loginError);
      setLoading(false);
    }
  }, [props.loginError]);

  const onClickLogin = () => {
    setLoading(true);
    setLoginError(undefined);
    props.onPressLoginBtn({
      handle: handle,
      token: token,
    });
  };

  return (
    <Surface style={styles.page}>
      <TextInput
        label="your access token"
        secureTextEntry
        onChangeText={text => setToken(text)}
        disabled={loading}
        error={!!loginError}
        defaultValue={props.userCredentials.token}
        mode="outlined"
        style={styles.textInput}
        accessibilityLabel="your access token"
      />
      <TextInput
        label="your handle"
        onChangeText={text => setHandle(text)}
        disabled={loading}
        error={!!loginError}
        defaultValue={props.userCredentials.handle}
        mode="outlined"
        style={styles.textInput}
        accessibilityLabel="your handle"
      />
      <OnlyShow If={!!loginError}>
        <View>
          <Paragraph
            accessibilityLabel="login status"
            style={styles.loginErrorText}>
            Login failed!
          </Paragraph>
          <Paragraph
            accessibilityLabel="login sub-status"
            style={[styles.loginErrorText, styles.loginErrorSubText]}>
            please check{' '}
            {loginError === 'AUTH_ERROR' ? 'credentials' : 'network connection'}{' '}
            and try again
          </Paragraph>
        </View>
      </OnlyShow>
      <Button
        onPress={onClickLogin}
        loading={loading}
        disabled={loginBtnDisabled}
        accessibilityLabel="login"
        style={styles.loginButton}>
        Login
      </Button>
    </Surface>
  );
}

const styleSheet = (theme: ThemeType) =>
  StyleSheet.create({
    page: {
      width: '100%',
      height: '100%',
      justifyContent: 'center',
      backgroundColor: theme.color.secondary,
    },
    textInput: {marginHorizontal: 20},
    loginErrorText: {color: 'red', textAlign: 'center'},
    loginErrorSubText: {fontStyle: 'italic'},
    loginButton: {
      backgroundColor: theme.color.primary,
      marginHorizontal: 50,
      marginVertical: 10,
    },
  });
