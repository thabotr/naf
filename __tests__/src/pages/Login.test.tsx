import React, {ReactNode} from 'react';
import renderer from 'react-test-renderer';
import {ThemeProvider} from '../../../src/context/theme';
import {Login} from '../../../src/pages/LoginPage';

const themed = (child: ReactNode) => <ThemeProvider>{child}</ThemeProvider>;

describe('Login page', () => {
  test('renders correctly when empty credentials and no error', () => {
    const tree = renderer
      .create(
        themed(
          <Login
            userCredentials={{handle: '', token: ''}}
            onPressLoginBtn={_ => {}}
          />,
        ),
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
