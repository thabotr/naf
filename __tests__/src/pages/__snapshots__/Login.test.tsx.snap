// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page renders correctly when empty credentials and no error 1`] = `
<View
  collapsable={false}
  style={
    {
      "backgroundColor": "#dbdbdb",
      "height": "100%",
      "justifyContent": "center",
      "shadowColor": "#000000",
      "shadowOffset": {
        "height": 3,
        "width": 0,
      },
      "shadowOpacity": 0.24,
      "shadowRadius": 4,
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "marginHorizontal": 20,
      }
    }
  >
    <View
      pointerEvents="none"
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 6,
          },
          {
            "backgroundColor": "#f6f6f6",
            "borderColor": "rgba(0, 0, 0, 0.54)",
            "borderRadius": 4,
            "borderWidth": 1,
          },
        ]
      }
      testID="text-input-outline"
    />
    <View>
      <View
        style={
          [
            {
              "paddingBottom": 0,
            },
            {
              "minHeight": 64,
              "paddingTop": 8,
            },
          ]
        }
      >
        <View
          collapsable={false}
          pointerEvents="none"
          style={
            {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": [
                {
                  "translateX": 0,
                },
              ],
              "zIndex": 3,
            }
          }
        >
          <View
            collapsable={false}
            pointerEvents="none"
            style={
              {
                "backgroundColor": "#f6f6f6",
                "bottom": 4,
                "left": 10,
                "maxHeight": 2,
                "opacity": 0,
                "position": "absolute",
                "right": 0,
                "top": 6,
                "transform": [
                  {
                    "translateX": 0,
                  },
                ],
                "width": 12,
              }
            }
          />
          <Text
            collapsable={false}
            maxFontSizeMultiplier={1.5}
            numberOfLines={1}
            style={
              {
                "backgroundColor": "#f6f6f6",
                "color": "transparent",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 18,
                "maxWidth": -28,
                "opacity": 0,
                "paddingHorizontal": 0,
                "position": "absolute",
                "textAlign": "left",
                "top": 37,
                "transform": [
                  {
                    "translateX": 0,
                  },
                  {
                    "translateY": 0,
                  },
                  {
                    "scale": 1,
                  },
                  {
                    "scaleY": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
          >
            your access token
          </Text>
          <Text
            collapsable={false}
            maxFontSizeMultiplier={1.5}
            numberOfLines={1}
            onLayout={[Function]}
            style={
              {
                "color": "#6200ee",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 0,
                "opacity": 0,
                "paddingHorizontal": 14,
                "position": "absolute",
                "textAlign": "left",
                "top": 36,
                "transform": [
                  {
                    "translateX": 0,
                  },
                  {
                    "translateY": 0,
                  },
                  {
                    "scale": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
          >
            your access token
          </Text>
          <Text
            collapsable={false}
            maxFontSizeMultiplier={1.5}
            numberOfLines={1}
            style={
              {
                "color": "rgba(0, 0, 0, 0.54)",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 0,
                "opacity": 0,
                "paddingHorizontal": 14,
                "position": "absolute",
                "textAlign": "left",
                "top": 36,
                "transform": [
                  {
                    "translateX": 0,
                  },
                  {
                    "translateY": 0,
                  },
                  {
                    "scale": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
          >
            your access token
          </Text>
        </View>
        <TextInput
          accessibilityLabel="your access token"
          defaultValue=""
          editable={true}
          maxFontSizeMultiplier={1.5}
          multiline={false}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder=" "
          placeholderTextColor="rgba(0, 0, 0, 0.54)"
          secureTextEntry={true}
          selectionColor="#6200ee"
          style={
            [
              {
                "flexGrow": 1,
                "margin": 0,
                "paddingHorizontal": 14,
                "zIndex": 1,
              },
              {
                "height": 56,
              },
              {
                "paddingBottom": 0,
                "paddingTop": 0,
              },
              {
                "color": "#000000",
                "fontFamily": "System",
                "fontSize": 16,
                "fontVariant": undefined,
                "fontWeight": undefined,
                "textAlign": "left",
                "textAlignVertical": "center",
              },
              false,
              [
                {},
              ],
            ]
          }
          testID="text-input-outlined"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
  </View>
  <View
    style={
      {
        "marginHorizontal": 20,
      }
    }
  >
    <View
      pointerEvents="none"
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 6,
          },
          {
            "backgroundColor": "#f6f6f6",
            "borderColor": "rgba(0, 0, 0, 0.54)",
            "borderRadius": 4,
            "borderWidth": 1,
          },
        ]
      }
      testID="text-input-outline"
    />
    <View>
      <View
        style={
          [
            {
              "paddingBottom": 0,
            },
            {
              "minHeight": 64,
              "paddingTop": 8,
            },
          ]
        }
      >
        <View
          collapsable={false}
          pointerEvents="none"
          style={
            {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": [
                {
                  "translateX": 0,
                },
              ],
              "zIndex": 3,
            }
          }
        >
          <View
            collapsable={false}
            pointerEvents="none"
            style={
              {
                "backgroundColor": "#f6f6f6",
                "bottom": 4,
                "left": 10,
                "maxHeight": 2,
                "opacity": 0,
                "position": "absolute",
                "right": 0,
                "top": 6,
                "transform": [
                  {
                    "translateX": 0,
                  },
                ],
                "width": 12,
              }
            }
          />
          <Text
            collapsable={false}
            maxFontSizeMultiplier={1.5}
            numberOfLines={1}
            style={
              {
                "backgroundColor": "#f6f6f6",
                "color": "transparent",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 18,
                "maxWidth": -28,
                "opacity": 0,
                "paddingHorizontal": 0,
                "position": "absolute",
                "textAlign": "left",
                "top": 37,
                "transform": [
                  {
                    "translateX": 0,
                  },
                  {
                    "translateY": 0,
                  },
                  {
                    "scale": 1,
                  },
                  {
                    "scaleY": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
          >
            your handle
          </Text>
          <Text
            collapsable={false}
            maxFontSizeMultiplier={1.5}
            numberOfLines={1}
            onLayout={[Function]}
            style={
              {
                "color": "#6200ee",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 0,
                "opacity": 0,
                "paddingHorizontal": 14,
                "position": "absolute",
                "textAlign": "left",
                "top": 36,
                "transform": [
                  {
                    "translateX": 0,
                  },
                  {
                    "translateY": 0,
                  },
                  {
                    "scale": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
          >
            your handle
          </Text>
          <Text
            collapsable={false}
            maxFontSizeMultiplier={1.5}
            numberOfLines={1}
            style={
              {
                "color": "rgba(0, 0, 0, 0.54)",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 0,
                "opacity": 0,
                "paddingHorizontal": 14,
                "position": "absolute",
                "textAlign": "left",
                "top": 36,
                "transform": [
                  {
                    "translateX": 0,
                  },
                  {
                    "translateY": 0,
                  },
                  {
                    "scale": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
          >
            your handle
          </Text>
        </View>
        <TextInput
          accessibilityLabel="your handle"
          defaultValue=""
          editable={true}
          maxFontSizeMultiplier={1.5}
          multiline={false}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder=" "
          placeholderTextColor="rgba(0, 0, 0, 0.54)"
          selectionColor="#6200ee"
          style={
            [
              {
                "flexGrow": 1,
                "margin": 0,
                "paddingHorizontal": 14,
                "zIndex": 1,
              },
              {
                "height": 56,
              },
              {
                "paddingBottom": 0,
                "paddingTop": 0,
              },
              {
                "color": "#000000",
                "fontFamily": "System",
                "fontSize": 16,
                "fontVariant": undefined,
                "fontWeight": undefined,
                "textAlign": "left",
                "textAlignVertical": "center",
              },
              false,
              [
                {},
              ],
            ]
          }
          testID="text-input-outlined"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
  </View>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "#87cefa",
        "borderColor": "transparent",
        "borderRadius": 4,
        "borderStyle": "solid",
        "borderWidth": 0,
        "elevation": 0,
        "marginHorizontal": 50,
        "marginVertical": 10,
        "minWidth": 64,
        "shadowColor": "#000000",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
      }
    }
  >
    <View
      accessibilityLabel="login"
      accessibilityRole="button"
      accessibilityState={
        {
          "disabled": true,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "overflow": "hidden",
          },
          {
            "borderRadius": 4,
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            [
              {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              {
                "textAlign": "left",
              },
              [
                {
                  "letterSpacing": 1,
                  "marginHorizontal": 16,
                  "marginVertical": 9,
                  "textAlign": "center",
                },
                undefined,
                {
                  "textTransform": "uppercase",
                },
                {
                  "color": "rgba(0, 0, 0, 0.32)",
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                {
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                undefined,
              ],
            ]
          }
        >
          Login
        </Text>
      </View>
    </View>
  </View>
</View>
`;
